packages/core-app/src/components/GraphicsCore/GraphicsCore.tsx

```diff
@@ -19,37 +19,68 @@ export const GraphicsCore: React.FC<GraphicsCoreProps> = ({
 	hideWelcomeFunc
 }) => {
 	const canvasEl = useRef<HTMLCanvasElement | null>(null);
+	const canvasEl2 = useRef<HTMLCanvasElement | null>(null);
 
 	useEffect(() => {
 		let canvasUpdatingInterval: NodeJS.Timeout;
 
 		if (canvasEl) {
 			const canvasElCurrent = canvasEl.current;
-			if (!canvasElCurrent) {
+			const canvasEl2Current = canvasEl2.current;
+
+			if (!canvasElCurrent || !canvasEl2Current) {
 				return;
 			}
 
 			const context = canvasElCurrent.getContext('2d');
+			const context2 = canvasEl2Current.getContext('2d');
 
-			if (!context) {
+			if (!context || !context2) {
 				return;
 			}
 
-			cvHelper.setCanvasElCurrent(canvasElCurrent);
+			cvHelper.setCanvasElCurrent(canvasElCurrent, canvasEl2Current);
 
 			canvasElCurrent.addEventListener("mousedown", (event) => {
 				if (!hideWelcome) {
 					hideWelcomeFunc(true);
 				}
-				const pixelRGB = GraphicsManipulator.getPixelRGB(context, canvasElCurrent, event);
+				const pixelRGB = GraphicsManipulator.getPixelRGBMouse(context, canvasEl2Current, event);
 				const hsvSelectColor = GraphicsManipulator.RGBtoHSV(pixelRGB);
 				cvHelper.setHSVSelectColor(hsvSelectColor);
 			});
 
+			canvasElCurrent.addEventListener("touchend", (event) => {
+				if (!hideWelcome) {
+					hideWelcomeFunc(true);
+				}
+				const pixelRGB = GraphicsManipulator.getPixelRGBTouch(context, canvasEl2Current, event);
+				const hsvSelectColor = GraphicsManipulator.RGBtoHSV(pixelRGB);
+				cvHelper.setHSVSelectColor(hsvSelectColor);
+			})
+
 			canvasUpdatingInterval = setInterval(async () => {
 				let toDraw = cvHelper.getLatestVideoFrame();
+
 				context.drawImage(toDraw, 0, 0, window.innerWidth, window.innerHeight * HEIGHT_MULTIPLIER);
-				cvHelper.draw(context, canvasElCurrent);
+
+				// clone context 1 to context 2 but contrasted
+				const imageData = context.getImageData(0, 0, window.innerWidth, window.innerHeight * HEIGHT_MULTIPLIER);
+
+				const clonedImageData = new ImageData(
+					new Uint8ClampedArray(imageData.data),
+					imageData.width,
+					imageData.height
+				);
+
+				GraphicsManipulator.contrastImage(clonedImageData, 2.5);
+
+				// uncomment to see live contrast data
+				context2.putImageData(clonedImageData, 0, 0);
+				const getNewImageData = context2.getImageData(0, 0, window.innerWidth, window.innerHeight * HEIGHT_MULTIPLIER);
+				context.putImageData(getNewImageData, 0, 0);
```

Comment:
even with commenting this line it doesnt work. It only works if you leave this line in.